lightning/backend/local.go

```diff
@@ -1122,19 +1122,27 @@ func (local *local) ImportEngine(ctx context.Context, engineUUID uuid.UUID) erro
 		log.L().Info("start import engine", zap.Stringer("uuid", engineUUID),
 			zap.Int("ranges", len(ranges)))
 
-		// split region by given ranges
-		for i := 0; i < maxRetryTimes; i++ {
-			err = local.SplitAndScatterRegionByRanges(ctx, ranges)
-			if err == nil {
-				break
-			}
+		// if all the kv can fit in one region, skip split regions. TiDB will split one region for
+		// the table when table is created.
+		if lf.TotalSize > local.regionSplitSize || lf.Length > regionMaxKeyCount {
+			// split region by given ranges
+			for i := 0; i < maxRetryTimes; i++ {
+				err = local.SplitAndScatterRegionByRanges(ctx, ranges)
+				if err == nil {
+					break
+				}
 
-			log.L().Warn("split and scatter failed in retry", zap.Stringer("uuid", engineUUID),
-				log.ShortError(err), zap.Int("retry", i))
-		}
-		if err != nil {
-			log.L().Error("split & scatter ranges failed", zap.Stringer("uuid", engineUUID), log.ShortError(err))
-			return err
+				log.L().Warn("split and scatter failed in retry", zap.Stringer("uuid", engineUUID),
+					log.ShortError(err), zap.Int("retry", i))
+			}
+			if err != nil {
+				log.L().Error("split & scatter ranges failed", zap.Stringer("uuid", engineUUID), log.ShortError(err))
+				return err
+			}
+		} else {
+			log.L().Info("engine total size is smaller than region split size, skip split regions",
+				zap.Stringer("uuid", engineUUID), zap.Int64("size", lf.TotalSize),
+				zap.Int64("count", lf.Length))
```

Comment:
should we still scatter 🤔

Reply:
I think not since we don't create any new region, pd should be responsible for the distribution of the existed regions.

Reply:
if we create lots of small tables on a fresh cluster will this cause all data still places on the same 3 stores? or will CREATE TABLE scatter?

Reply:
Yes. Maybe we shall still scatter it.

Reply:
Currently `preSplitAndScatter` depend on the session variable `tidb_scatter_region` (default is 'OFF') for whether scatter or not. So maybe we can set this variable to 'ON' at restoreSchema. 
See: https://github.com/pingcap/tidb/blob/46fa1628896fce4e30d990b9508479bc7e3add2b/ddl/ddl_api.go#L1869-L1893

Reply:
@glorv SGTM.

Reply:
There is still a problem, the `tidb_scatter_region` is a global variable, so if we switch it to off before restoring schemas and on after finished, how to ensure it is restored if lightning exit before finishing restoreSchema. 

I added a feature request to tidb. https://github.com/pingcap/tidb/issues/21991

Reply:
@glorv what if we add `PRE_SPLIT_REGIONS=1` to the `CREATE TABLE` statement